ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/group.j2

```diff
@@ -0,0 +1,36 @@
+{% if group.bgp is arista.avd.defined %}
+{%   for group_name in group.bgp | arista.avd.natural_sort %}
+!
+group bgp {{ group_name }}
+{# EOS cli lists vrf before neighbor #}
+{%     if group.bgp[group_name].vrf is arista.avd.defined %}
+   vrf {{ group.bgp[group_name].vrf }}
+{%     endif %}
+{%     if group.bgp[group_name].neighbor is arista.avd.defined %}
+{%       for neighbor in group.bgp[group_name].neighbor %}
+   neighbor {{ neighbor }}
+{%       endfor %}
+{%     endif %}
+{%     if group.bgp[group_name].maintenance_profile.bgp is arista.avd.defined %}
+   maintenance profile bgp {{ group.bgp[group_name].maintenance_profile.bgp }}
+{%     endif %}
+{%   endfor %}
+{% endif %}
+{% if group.interface is arista.avd.defined %}
+{%   for group_name in group.interface | arista.avd.natural_sort %}
+!
+group interface {{ group_name }}
+{%     if group.interface[group_name].interfaces is arista.avd.defined %}
+{%       for interface in group.interface[group_name].interfaces %}
```

Comment:
you can consider using `arista.avd.natural_sort` to simplify the template, since it covers the test above.